### DISCUSSION TITLE
Coming from Airtable, how do I do this?

### DISCUSSION THREAD

**Discussion by charleslcso** [Budibase/budibase#9986] [Help] [answered]
Dear experts,

I need to know if Budibase can perform these function urgently. Ideally I need to know if I can achieve this using no-code, or I need to write some/a lot of custom code. 

I've captured a short movie. Please help me by viewing it and let me know. The video is 14MB in size, so please click on the link here.

https://www.tourdefarm.com/budibase.mov

**Comment by ConorWebb96** [comment, marked as answer]
Hey @charleslcso,

I've had a look at the video, yes this is possible with Budibase. However it is highly dependant on the Whatsap API. I can see it does supply nice hooks and is fairly well documented. The messages information it returns also seems to be formatted reasonable well and would work with the Budibase components.

I would be confident in saying that you could pull together the desired effect. Obviously this wouldn't work out of the box, it would require API integration to get the customer numbers and names.

**Side note**
This could be achieved with https://faq.whatsapp.com/425247423114725/?helpref=uf_share&cms_platform=iphone, you would still need to get the numbers. 

This along with the dynamic formula column would be able to achieve something like this, all you would again need is to append the number with the message (which would be encoded like the example shown in the link above.) From here you would just add a button click event that would trigger this link.

**Reply by charleslcso** [reply]
Connor, thank you for your reply! Much appreciated 

The click of button to send WhatsApp message doesn’t call any API; that’s the exact thing we try to avoid. I’m sure you know a few reasons for this. 

ummm. I need to do a POC first. 

How about data, in this case it’s the Contacts created/imported by users, can stay / view by that user? For example, User A has 10 contacts (rows in table), User B has 20 contacts. Each user’s data is private to himself. 

Possible?

**Reply by melohagan** [reply]
Hey @charleslcso 

You can POST to an API on button click via the **Execute Query** action: https://docs.budibase.com/docs/data-sources#tutorial-posting-form-data-to-a-custom-query

You can also display data that is related to the current logged in user like so: https://docs.budibase.com/docs/fetching-data#display-user-specific-data

**Reply by ConorWebb96** [reply]
Hey @charleslcso, 

I can confirm that this will indeed work with Budibase.

I have some images below just showing the process made for this to work.

**Tables**
![dataTable](https://user-images.githubusercontent.com/126772285/224968053-bbed033b-9bf6-43af-af96-b57e4760c9b6.png)
<img width="1581" alt="customerTable" src="https://user-images.githubusercontent.com/126772285/224968218-5800acb4-94df-4d6d-8ee6-93d3e94c063a.png">

**Create messages**
<img width="401" alt="Create message" src="https://user-images.githubusercontent.com/126772285/224968448-a62df383-f41d-4698-a6e9-c785db8199c0.png">

**Message Table**
<img width="1345" alt="table after creation" src="https://user-images.githubusercontent.com/126772285/224968508-976bab5c-9aa9-47a2-a48b-a5a497441f41.png">

**Customer**
![customer creation window](https://user-images.githubusercontent.com/126772285/224968550-e3706187-4947-4e4d-a180-7a0fb2f9f76b.png)
![table with button action](https://user-images.githubusercontent.com/126772285/224968597-4ebcbb59-bedc-4296-b153-70661f2a2288.png)

**Prompt after clicking button**
![whatsappprompt](https://user-images.githubusercontent.com/126772285/224968673-9c9d2282-2d3d-447f-88cb-fb41f47db823.png)

**Whats app message being sent**
![whatsapp](https://user-images.githubusercontent.com/126772285/224968825-7788709a-66ad-413f-a4f2-3a097e10846c.png)

**Table fields**
<img width="1581" alt="customerTable" src="https://user-images.githubusercontent.com/126772285/224968957-8d89c2ba-aef1-4974-bc5c-832e26ae93fd.png">
![dataTable](https://user-images.githubusercontent.com/126772285/224968960-822f9ffa-e1f9-4c0d-bdb1-c0c4647d25f8.png)

**Code combined with fomula column to get it to work**
<img width="501" alt="Screenshot 2023-03-14 at 10 15 01" src="https://user-images.githubusercontent.com/126772285/224969104-5ad5f1a8-e2dc-4fcd-9c63-ab1b634dfbfd.png">

Hopefully this helps!
